views/default/input/submit.php

```diff
@@ -6,10 +6,16 @@
  * @subpackage Core
  *
  * @uses $vars['class'] Additional CSS class
+ * @uses $vars['text']  Text of the submit button
+ * @uses $vars['value'] Value of the submit input
  */
 
 $vars['type'] = 'submit';
 
 $vars['class'] = elgg_extract_class($vars, 'elgg-button-submit');
 
+if (!isset($vars['text']) && isset($vars['value'])) {
+	$vars['text'] = $vars['value'];
+}
```

Comment:
No, you want the `value` attribute of the button if it's set and someone passes `name` as well.